material-ui/material-ui.d.ts

```diff
@@ -7228,7 +7228,7 @@ declare module 'material-ui/utils/withWidth' {
       mediumWidth?: number;
       resizeInterval?: number;
   }
-  export default function withWidth(options?: Options): __React.ComponentClass<any>
+  export default function withWidth<C extends __React.ComponentClass<any> | __React.StatelessComponent<any>>(options?: Options): (component: C) => C;
```

Comment:
I'm not quite sure what do you mean by `constructor type`. The idea is that it returns the same component with one more property `width` that can be used in the wrapped component.

BTW the React typings are missing an abstraction for HOC.